### DISCUSSION TITLE
wtf/crud: DialService instead of wtf.Dialer

### DISCUSSION THREAD

**Discussion by komuw** [benbjohnson/wtf#14] [Q&A] [answered]
https://www.gobeyond.dev/crud/   
Generally in Go, interfaces are named with an `er`; `io.Reader` for reading `io.Writer` for writing etc.    
In `wtf` we have interfaces like `DialService`. Is there a reason for deviating from the norm? I would have expected it to be called `wtf.Dialer`

**Comment by benbjohnson** [comment, marked as answer]
The `-er` suffix works well for simple interfaces like `io.Reader` & `io.Writer` (or even `io.ReadCloser`). One option is to break up the interface into smaller interfaces like `DialFinder`, `DialCreator`, `DialUpdater`, & `DialDeleter`. However, I've found that you end up with a lot of tiny interfaces without much benefit. 

Naming the service as `Dialer` is an option but that would imply that there's a `Dial()` function. That's the case with [`net.Dialer`](https://golang.org/pkg/net/#Dialer) (although that's not an interface).

Ultimately it comes down to what works for me in practice. "Service" is generic enough that I don't need to think about what to name the CRUD interface but it's clear that it handles an entity.
